rosbridge_library/src/rosbridge_library/capability.py

```diff
@@ -68,6 +70,11 @@ def __init__(self, protocol: Protocol) -> None:
         """
         self.protocol = protocol
 
+        if self.parameter_names and self.protocol.parameters:
+            for param_name in self.parameter_names:
```

Comment:
This and the next line could be a single list comprehension

Reply:
Could you show what specific you have in mind?

Reply:
... actually I just noticed another option

```python
for param_name, param_value in self.parameter_names.items():
    if param_name in self.protocol.parameters:
        setattr(self, param_name, param_value)
```

this isn't a major comment, i was just staring at python

Reply:
Is this actually a thing? I haven't seen such syntax in python

Reply:
I just tested:
```
for name in list1 if name in list2:
    print(name) 
```
And it gave me syntax error

Reply:
ugh no, I missed a thing. It would actually be this, which already makes it too hard to read

```python
for param_name in [p for p in self.parameter_names if p in self.protocol.parameters:

Reply:
Nah, this starts to look ugly. I might go with your other proposition though

Reply:
> ... actually I just noticed another option
> 
> ```python
> for param_name, param_value in self.parameter_names.items():
>     if param_name in self.protocol.parameters:
>         setattr(self, param_name, param_value)
> ```
> 
> this isn't a major comment, i was just staring at python

I cannot use `items()` on `parameter_names` as it is a tuple of strings, not a dictionary. This would have to iterate over `parameters` instead:
```Python
        if self.parameter_names and self.protocol.parameters:
            for param_name, param_value in self.protocol.parameters.items():
                if param_name in self.parameters_names:
                    setattr(self, param_name, param_value)
```